tests/test_configurationmanager.py

```diff
@@ -0,0 +1,54 @@
+import tests.context
```

Comment:
1. I disagree. How do we test `run_setup_mode` without creating a new config file?

2. Actually (might just be a pytest thing) creating a new instance for each test triggers the singleton exception. But yes, I can just use fixtures.

3. Agreed. Would probably accomplish this with fixtures, if I understand them correctly.

Reply:
1. There is no need to delete the config file outside of testing. To test `run_setup_mode` you could set `config_filepath` to another path inside of the artifacts folder and clean up right after the test. Since this is only relevant to testing, this clean up should be part of the tests (not the class).

2. Does this happen even if you call `getInstance()` in the fixture?

Reply:
1. I think I see what you're saying now. So here's what I'm thinking:
* Delete `core/configuration.ini` (since it's only needed at the start for testing).
* Create `tests/artifacts/configuration.ini` so that every other test can use that for testing.
* Create `tests/artifacts/config_manager` so that the configuration manager can create config files in this folder for testing (and the folders will be removed as a part of testing, not a method that's a part of `ConfigurationManager`.

2. No, since a new instance would not be created each test. So yes, we can use `getInstance()` in the fixture.

I'll update this thread when I make the necessary changes.

Reply:
Sounds good.

Reply:
Made these changes. Decided that `getInstance` (now `get_instance to keep everything camel case) should take parameters updating the current config file location and input function (which differ among tests).

Reply:
It seems we were working in parallel, but I made some comments in regards to the new get_instance. I think you need to go to "file view" to see them.